pkg/discovery/worker/aggregation/container_usage_data_aggregator.go

```diff
@@ -0,0 +1,84 @@
+package aggregation
+
+import (
+	"fmt"
+	"github.com/turbonomic/turbo-go-sdk/pkg/proto"
+	"math"
+)
+
+var (
+	avgUsageDataStrategy = "avgUsageData"
+	maxUsageDataStrategy = "maxUsageData"
+
+	DefaultContainerUsageDataAggStrategy = avgUsageDataStrategy
+
+	// Map from the configured utilization data aggregation strategy to utilization data aggregator
+	ContainerUsageDataAggregators = map[string]ContainerUsageDataAggregator{
+		avgUsageDataStrategy: &avgUsageDataAggregator{aggregationStrategy: "average usage data strategy"},
+		maxUsageDataStrategy: &maxUsageDataAggregator{aggregationStrategy: "max usage data strategy"},
+	}
+)
+
+// ContainerUsageDataAggregator interface represents a type of container usage data aggregator
+type ContainerUsageDataAggregator interface {
+	// AggregationStrategy returns aggregation strategy of this data aggregator
+	AggregationStrategy() string
```

Comment:
I am not sure how you plan to use the `AggregationStrategy()`. However with the current usage of this used only in print/log statements, you can implement a `String()` method instead and you can use the structure directly and the fmt (or other loggers) can decipher if the structure implements `String()`.
Example ref: https://play.golang.org/p/uQL_RMJ_Bj.

If you need `AggregationStrategy()` for a different reason, you can use it to store and return a string enum, something like what you have created a constant above `avgUsageDataStrategy = "avgUsageData"`.
After redefining it as:
```
type DataStrategy string
const (
    avgUsageDataStrategy DataStrategy = "avgUsageData"
    ............
)```

Reply:
I plan to use the `AggregationStrategy()` only for log statements to show the aggregation strategy.

Here I'm using strategy pattern to create 2 different aggregation strategies:

Reply:
@marvelyue Apologies, I think my usage of word structure confused you. I understand your code.
To be more clear.
What I meant is you can do this:
Update your interface to 
```
type ContainerUsageDataAggregator interface {
	String() string
	Aggregate(containerCommodities []*proto.CommodityDTO) (float64, float64, float64, error)
}
```
Implementation of `String()` by the structure does what `AggregationStrategy()` does.
and whereever you use the same to print you can use the interface variable directly, for example:
```
usageDataAggregator, _ := agg.ContainerUsageDataAggregators[usageDataAggStrategy]

glog.Infof("ContainerSpec will aggregate Containers usage data by %s", usageDataAggregator)
```

 and the fmt (or other logger) will decipher that your interface has a stringer and use it.

A much simpler reference [here](https://tour.golang.org/methods/17)

Reply:
Ahh, I see what you meant. Yes it makes perfect sense to use `String()` function to replace `AggregationStrategy()` function for printing logs.

Thanks for the suggestion. I'll make the changes in this following PR https://github.com/turbonomic/kubeturbo/pull/411